### ISSUE TITLE
Top Foreign Languages Analysis

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/ML-Crate#519]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Top Foreign Languages Analysis
:red_circle: **Aim** : The aim of this project is to analyze the top foreign languages based on the given dataset.
:red_circle: **Dataset** : https://www.kaggle.com/datasets/timmofeyy/top-foreign-languages-preply-tutors
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by Yuvika-14** [comment]
hi abhisekh
Full name: Yuvika Singh
git hub profile link:https://github.com/Yuvika-14/Yuvikademo.git
approch: ensemble methods, gradient boosting, neural networks.
Take care of the missing data if there are any categorical values then will use one hot encoder or label encoder depending upon the issue.

**Comment by abhisheks008** [comment]
Implement these models for this project,
1. Random Forest
2. Decision Tree
3. Logistic Regression
4. XgBoost
5. Gradient Boosting
6. MLP
7. Lasso
8. Ridge

Issue assigned to you @Yuvika-14

**Comment by abhisheks008** [comment]
Issue assigned to you @ghousiya47

**Comment by abhisheks008** [comment]
Hi @ghousiya47 as this a machine learning project with proper implementation, I know it requires time to put forward the highest quality in the project. You can take as much time as you, but make sure you are creating something which should be of highest quality.

**Comment by abhisheks008** [comment]
No issues with the semester exams, I hope you have done well in that. You can start contributing now.

You need to find out the best csv file which directly suggests the problem statement given to you. You can use all the csv files, or you can use one of them, or even you can use 2-3 of them, it depends on you and your requirement.

Model creation is for implementing machine learning methods to a certain parameter which you might think is matching with the problem statement. You need to explore the dataset and try to find out that.

@ghousiya47
